collections/src/main/scala/strawman/collection/immutable/ChampHashMap.scala

```diff
@@ -0,0 +1,689 @@
+package strawman
+package collection.immutable
+
+import collection.{IterableFactoryLike, Iterator, MapFactory, StrictOptimizedIterableOps}
+import collection.Hashing.{computeHash, keepBits}
+
+import scala.annotation.unchecked.{uncheckedVariance => uV}
+import scala.{Any, AnyRef, Array, Boolean, Double, IndexOutOfBoundsException, Int, NoSuchElementException, None, Nothing, Option, SerialVersionUID, Serializable, Some, Tuple2, Unit, `inline`, math, sys, UnsupportedOperationException}
+import scala.Predef.{assert, require, ???}
+
+import java.lang.Integer.bitCount
+import java.lang.Math.ceil
+import java.lang.System.arraycopy
+
+import strawman.collection.mutable.{Builder, ImmutableBuilder}
+
+/** This class implements immutable maps using a Compressed Hash-Array Mapped Prefix-tree.
+  * See paper https://michael.steindorfer.name/publications/oopsla15.pdf for more details.
+  *
+  *  @tparam K      the type of the keys contained in this hash set.
+  *  @tparam V      the type of the values associated with the keys in this hash map.
+  *
+  *  @author  Michael J. Steindorfer
+  *  @version 2.13
+  *  @since   2.13
+  *  @define Coll `immutable.ChampHashMap`
+  *  @define coll immutable champ hash map
+  */
+
+@SerialVersionUID(2L)
+final class ChampHashMap[K, +V] private[immutable] (val rootNode: MapNode[K, V], val cachedJavaHashCode: Int, val cachedSize: Int)
+  extends Map[K, V]
+    with MapOps[K, V, ChampHashMap, ChampHashMap[K, V]]
+    with StrictOptimizedIterableOps[(K, V), Iterable /* ChampHashMap */, ChampHashMap[K, V]]
+    with Serializable {
+
+  def iterableFactory: IterableFactoryLike[List] = List
```

Comment:
I would use the default `immutable.Iterable` here:

~~~ scala
def iterableFactory: IterableFactory[Iterable] = Iterable
~~~

(currently the default `Iterable` happens to be `List` but that could change…)